File: hashing/java/LongestSubstrWOutRep.java
public class Solution {
    public int lengthOfLongestSubstring(String A) {
        int n=A.length();
        if(n==0||n==1)return 1;
        int i=0, ans=1;
        //HashSet<Integer>S= new HashSet<>();
        HashMap<Character, Integer> M= new HashMap<>(); 
        //S.add(A.charAt(0));
        M.put(A.charAt(0), 0);
        
        for(int j=1;j<n;j++)
        {
            //S.add(A.charAt(j));
            if(M.containsKey(A.charAt(j))==true)
            {
                while(i<n&&A.charAt(i)!=A.charAt(j))
                {
                    M.remove(A.charAt(i)); i++;
                }
                if(i>=n)return ans;
                M.remove(A.charAt(i));i++;
            }
            else ans=Math.max(ans, j-i+1);
            M.put(A.charAt(j), j);
        }
        return ans;
    }
}

File: hashing/C++/longestSubstrWOutRep.cpp
unordered_map<char,int> m;
    int j=0,i=0, n =A.size(), ans=0;
    
    for(int i=0;i<n;i++){
        if(!m[A[i]]){}
        else j=max(j,m[A[i]]); 
        
        m[A[i]]=i+1;
        ans= max(ans, i-j+1);
    } 
    return ans; 